xenstore/xenstore.go

```diff
@@ -222,6 +225,53 @@ func xs_ls(script_name string, args []string) {
 	}
 }
 
+func xs_chmod(script_name string, args []string) {
+	if len(args) == 0 || args[0] == "-h" || len(args) < 2 {
+		die("Usage: %s key mode [modes...]", script_name)
+	}
+
+	var err error
+	key := args[0]
+	var perms []xenstoreclient.Permission
+
+	for _, m := range args[1:] {
+		if len(m) > 1 {
+			var p xenstoreclient.Permission
+			switch m[0] {
+			case 'n':
+				p.Pe = xenstoreclient.PERM_NONE
+			case 'r':
+				p.Pe = xenstoreclient.PERM_READ
+			case 'w':
+				p.Pe = xenstoreclient.PERM_WRITE
+			case 'b':
+				p.Pe = xenstoreclient.PERM_READWRITE
+			default:
+				err = errors.New("Invalid mode string")
+			}
+			if err == nil {
+				var id uint64
+				id, err = strconv.ParseUint(m[1:], 10, 0)
```

Comment:
what is the magic number `10`? is it possible to assign the value to  a meaningful name?

Reply:
10 is the base for conversion.
it's OK to use it in this way:
see this example in std lib:
https://github.com/golang/go/blob/master/src/strconv/atoi.go#L247